mixer/adapter/stackdriver/cmd/main.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// THIS FILE IS AUTOMATICALLY GENERATED.
```

Comment:
Is it really?

Reply:
Yes. It is. https://github.com/istio/istio/pull/9362/files#diff-6bbb65fbbd18aef0a069f136274548adR3

Reply:
I see. Perhaps, we could use this tool as a one-off run to convert all in-proc adapters to out-of-proc adapters. If that's the plan, then we need to make sure it works OK for other high-importance candidates to convert to OOP (prometheus?). @douglas-reid @mandarjog WDYT?

Reply:
Yeah, that is why I use stackdriver as an example here for this tool's correctness review.

Reply:
Could you try to run an experiment to generate code for other adapters and see if they compile. It's very hard to tell from the template whether it's working OK. Then we can submit the stackdriver OOP first as a proof of it working, then the tool second.

Reply:
All adapters should compile with the generated code: https://github.com/istio/istio/pull/12357. (except kubeenv adapter, the tool does not have template for attributes generator adapters) Though several of them would not work properly and need some code change in adapter itself since they use inferred types in adapter building or data path -- namely stdio, statsd, stackdriver, signalfx (luckily, they seemed to migrate to [a oop adapter](https://github.com/signalfx/signalfx-istio-adapter) already).

I could extract out stackdriver oop generated code to another pr for review, and then we can come back to this pr for the tool. Thanks!